public/app/features/panel/panellinks/link_srv.ts

```diff
@@ -20,96 +80,92 @@ export class LinkSrv {
-        paramsArray.push(key);
-      } else if (_.isArray(value)) {
-        _.each(value, instance => {
-          paramsArray.push(key + '=' + encodeURIComponent(instance));
-        });
-      } else {
-        paramsArray.push(key + '=' + encodeURIComponent(value));
-      }
-    });
-
-    if (paramsArray.length === 0) {
-      return url;
-    }
-
-    return this.appendToQueryString(url, paramsArray.join('&'));
-  }
-
-  appendToQueryString(url, stringToAppend) {
-    if (!_.isUndefined(stringToAppend) && stringToAppend !== null && stringToAppend !== '') {
-      const pos = url.indexOf('?');
-      if (pos !== -1) {
-        if (url.length - pos > 1) {
-          url += '&';
-        }
-      } else {
-        url += '?';
-      }
-      url += stringToAppend;
-    }
-
-    return url;
-  }
-
-  getAnchorInfo(link) {
+  getAnchorInfo(link: any) {
     const info: any = {};
     info.href = this.getLinkUrl(link);
     info.title = this.templateSrv.replace(link.title || '');
     return info;
   }
 
-  getPanelLinkAnchorInfo(link, scopedVars) {
-    const info: any = {};
-    info.target = link.targetBlank ? '_blank' : '';
-    if (link.type === 'absolute') {
-      info.target = link.targetBlank ? '_blank' : '_self';
-      info.href = this.templateSrv.replace(link.url || '', scopedVars);
-      info.title = this.templateSrv.replace(link.title || '', scopedVars);
-    } else if (link.url) {
-      info.href = link.url;
-      info.title = this.templateSrv.replace(link.title || '', scopedVars);
-    } else if (link.dashUri) {
-      info.href = 'dashboard/' + link.dashUri + '?';
-      info.title = this.templateSrv.replace(link.title || '', scopedVars);
-    } else {
-      info.title = this.templateSrv.replace(link.title || '', scopedVars);
-      const slug = kbn.slugifyForUrl(link.dashboard || '');
-      info.href = 'dashboard/db/' + slug + '?';
-    }
-
-    const params = {};
-
-    if (link.keepTime) {
-      const range = this.timeSrv.timeRangeForUrl();
-      params['from'] = range.from;
-      params['to'] = range.to;
-    }
+  getDataPointVars = (seriesName: string, valueTime: DateTime) => {
+    const valueTimeQuery = toUrlParams({
+      time: dateTime(valueTime).valueOf(),
+    });
 
-    if (link.includeVars) {
-      this.templateSrv.fillVariableValuesForUrl(params, scopedVars);
-    }
+    const seriesQuery = toUrlParams({
+      series: seriesName,
+    });
 
-    info.href = this.addParamsToUrl(info.href, params);
+    return {
+      [DataLinkBuiltInVars.valueTime]: {
+        text: valueTimeQuery,
+        value: valueTimeQuery,
+      },
+      [DataLinkBuiltInVars.seriesName]: {
+        text: seriesQuery,
+        value: seriesQuery,
+      },
+    };
+  };
+
+  getDataLinkUIModel = (link: DataLink, scopedVars: ScopedVars, dataPoint?: LinkDataPoint) => {
+    const params: KeyValue = {};
+    const timeRangeUrl = toUrlParams(this.timeSrv.timeRangeForUrl());
+
+    const info: LinkModel = {
+      url: link.url,
```

Comment:
seems to be some typing issues here, in the old panels the expected return prop is href, so getting type errors due to that.